app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,436 @@
+     * @return the id of the message
+     */
+    public int message(String message, String actionMessage, View.OnClickListener actionOnClick,
+                       int priority){
+        return message(message, actionMessage, actionOnClick, priority, true, 0);
+    }
+
+    /**
+     * Add to the list and display a new message. This is the method that should be used to
+     * display messages and it's being called by {@link BaseActivity#snackBar}.
+     * @param message The message String.
+     * @param actionMessage The label for the action button.
+     * @param actionOnClick The View.OnClickListener for the action button.
+     * @param priority The priority of the message. The param is a int and the lower the number the
+     *                 higher priority the message has. 0 is the highest.
+     * @param isDismissible if true the message will have a X button to remove it self from the
+     *                      list.
+     * @param secondsToTimeOut Number of seconds to message auto dismiss. 0 to never.
+     * @return the id of the message.
+     */
+    public int message(String message, String actionMessage, View.OnClickListener actionOnClick,
+                        int priority, boolean isDismissible, int secondsToTimeOut){
+        mMessageId++;
+        MessageKey key = new MessageKey(mMessageId, priority);
+        Message value = new Message(key, message, actionMessage, actionOnClick, isDismissible);
+        mMessagesList.put(key, value);
+        updateList();
+        setTimer(key, secondsToTimeOut);
+        if (mMessagesList.size() == 0) {
+            hide();
+        } else {
+            show();
+        }
+        return mMessageId;
+    }
+
+    /**
+     * Sets the auto-dismiss timer to the message given it's key and seconds to dismiss.
+     * @param key The message key.
+     * @param seconds Seconds until dismiss.
+     */
+    private void setTimer(final MessageKey key, int seconds) {
+        if(seconds > 0) {
+            int limit = seconds*1000;
+            new CountDownTimer(limit, 1000) {
+                @Override public void onTick(long millisUntilFinished) {
+                }
+
+                @Override public void onFinish() {
+                    mMessagesList.remove(key);
+                    updateList();
+                }
+            }.start();
+        }
+    }
+
+    /**
+     * Programmatically dismiss a message by it's id.
+     * @param id The message id.
+     */
+    public void dismiss(int id) {
+        dismiss(getKey(id));
+    }
+
+    /**
+     * Programmatically dismiss a message by a MessageKey Object
+     * @param key The message Key.
+     */
+    public void dismiss(MessageKey key) {
+        mMessagesList.remove(key);
+        updateList();
+    }
+
+    /**
+     * Find message Key by it's id value.
+     * @param id The id of the message.
+     * @return The MessageKey of the message.
+     */
+    private MessageKey getKey(int id) {
```

Comment:
Actually I forgot to create a snackbarDIsmiss() function on BaseActivity. Now It's done.